src/main/kotlin/no/nav/syfo/infrastructure/database/repository/VedtakRepository.kt

```diff
@@ -143,12 +143,18 @@ class VedtakRepository(private val database: DatabaseInterface) : IVedtakReposit
             connection.commit()
         }
 
-    override fun getUnpublishedVedtakStatus(): List<Vedtak> =
+    override fun getUnpublishedVedtakStatus(): List<Pair<Vedtak, VedtakStatus>> =
         database.connection.use { connection ->
             connection.prepareStatement(GET_UNPUBLISHED_VEDTAK_STATUS).use {
                 it.executeQuery().toList { toPVedtak() }
-            }.map { pVedtak ->
-                pVedtak.toVedtak(connection.getVedtakStatus(pVedtak.id))
+            }.flatMap { pVedtak ->
+                val pVedtakStatusListe = connection.getVedtakStatus(pVedtak.id)
+                pVedtakStatusListe.filter { it.publishedAt == null }.map { pVedtakStatus ->
+                    Pair(
+                        pVedtak.toVedtak(pVedtakStatusListe),
```

Comment:
Hvordan blir det her egentlig? Vi får en `Pair` med vedtak som har sin egen statusliste og en upubliserte vedtakstatus (som også finnes i vedtak sin statusliste). Litt forvirrende, men kanskje det går greit :) 
Ville det blitt bedre om man bare returnerte en liste med Vedtak (som har upubliserte statuser) herfra og så kan servicen publisere upubliserte statuser fra vedtak sin statusliste?

Reply:
Ja, det kunne vært et alternativ. Men da må vi nesten ha med publishedAt i modellklassen for VedtakStatus.

Men ja, slik det er gjort her så er det underforstått at instansen av VedtakStatus hører til Vedtak'et.

Reply:
Ja, ikkesant :thinking: Ikke superviktig hvis det blir knotete, ble bare litt forvirrende for meg at vedtakstatusen på en måte lever dobbelt i strukturen her.